Add Boost Insurance case study (#20232)
pulumi/docs: content/case-studies/boost-insurance.md

---
title_tag: Boost Insurance | Case Studies
title: "How Boost Insurance Eliminates Data Breach Risk with Pulumi"
allow_long_title: true
description: |
    Boost Insurance built a self-service platform with Pulumi that provisions ephemeral, time-bound environments in the cloud, eliminating the risk of storing production data on developer machine.

> meta_desc: Boost Insurance eliminated production data on developer laptops by building a self-service platform with Pulumi, reducing breach risk and saving developer time.

Reviewer: We might want to use the phrases "regulatory compliance" (and also, if applicable - not sure - "data residency") as prominently as possible.

  headline_stat: Zero
  headline: production data on developer laptops, eliminating breach risk

exec_summary: |
    [Boost Insurance](https://www.boostinsurance.com/) operates in one of the most heavily regulated industries in the United States. To debug production issues, developers had been downloading database snapshots to their laptops, bringing hundreds of gigabytes of sensitive data onto their personal machines. Boost's platform engineering team used Pulumi's Automation API, Pulumi Deployments, and Pulumi ESC to build a self-service platform enabling developers to provision isolated, time-limited database environments entirely in the cloud, ensuring production data no longer leaves Boost's controlled infrastructure and saving developers roughly 30 minutes per debugging session.

sections:
    - label: Exec Summary
      anchor: executive-summary
    - label: The Challenge
      anchor: the-challenge
    - label: What Boost Built
      anchor: the-solution
    - label: Results
      anchor: results
    - label: How Pulumi Works
      anchor: how-pulumi-works
    - label: Conclusion
      anchor: conclusion
---

## The business challenge {#the-challenge}

> Boost Insurance operates in one of the most heavily regulated industries in the United States.

Reviewer: Should _technically_ be one line per paragraph for nice diffing purposes IMO, but it's a customer story and it doesn't matter that much.

Every policy the platform issues or binds — even a test policy — becomes a matter of record that
must be reported to regulatory bodies. That means production data can't be used for testing, and
reproducing production issues in a development environment is often impractical.

Developers needed access to production data to debug real-world issues, and the only option was
to download database snapshots to their laptops. For a 100 GB quoting database, that meant
45-minute downloads for remote team members, hundreds of gigabytes of sensitive data stored on
personal machines, and no guarantee that data stayed within the company's security perimeter.

"Someone goes and steals a developer's laptop — we now have a reportable breach," said Richard Genthner, CISO at Boost Insurance. "We have to report not only to our partners, but to the state of New York, all 50 states, and the federal government. We have to notify every single person that's ever had a policy with us. It becomes a massive problem."

Boost has developers working remotely across multiple countries, and some partner contracts

> explicitly require that data never leaves the United States. The existing tooling — StrongDM for

Reviewer: We may not want to call out StrongDM by name here since they are not a direct competitor of ours.

- **30 minutes saved per developer per debugging session.** Restoring a database in the cloud takes
about 15 minutes, compared to 45 or more minutes for a remote developer to download and restore
locally. Across Boost's development team, that adds up to significant recovered engineering time.

- **Self-service replaces an operational bottleneck.** Before the platform, developers had to wait
for the platform engineering team to manually provision database environments. Now developers
create, extend, and tear down isolated environments on their own, freeing up the platform team to
focus on higher-value work.

- **Audit compliance built in.** Every database connection is routed through IAP-protected tunnels
with full query logging, maintaining the audit trail that insurance regulators require — the same
capability that the previous tooling provided, now integrated directly into the platform.

## How Pulumi makes it work {#how-pulumi-works}

Boost selected Pulumi because its feature set aligned with the specific requirements of a
self-service, compliance-focused platform.

The [Automation API](/docs/iac/concepts/automation-api/) is the foundation. Rather than shelling

> out to `pulumi up`, Boost's Go application embeds Pulumi as a library, programmatically creating

Reviewer: Given that this is a customer story, we might want to make this a little less technical and talk more about what capabilities it enables, which is what we get at in the next sentence.

and managing stacks for each developer's isolated environment. This is what enabled the team to
build a polished self-service experience — a web UI, a CLI, and a desktop client — all backed by the
same infrastructure-as-code logic.

[Pulumi Deployments](/docs/deployments/) and its [TTL stacks](/docs/deployments/concepts/ttl/)
feature handle the environment lifecycle. Each isolated database is a TTL stack with a
configurable lifetime. When the TTL expires, Pulumi Deployments automatically destroys the stack
and all of its resources, no cron jobs or manual cleanup required. This is what ensures that

> isolated environments don't accumulate cost or become a security liability.

Reviewer: Again, suggest leading with the capabilities (cost, security) rather than the technical details.

[Pulumi ESC (Environments, Secrets, and Configuration)](/docs/esc/) manages credentials across
Boost's multi-project GCP environment using OIDC federation. Each GCP project has its own
short-lived credentials in ESC, and Pulumi Deployments consumes them automatically when
provisioning and destroying environments. This eliminates long-lived service account keys and

> keeps the credential lifecycle fully automated.

Reviewer: And same here.

[Pulumi Cloud](/docs/pulumi-cloud/) ties it all together with centralized visibility into every

> active environment, with stacks, resources, and update history in one place. For a regulated insurance

Reviewer: And again.

company, the ability to export audit logs from Pulumi Cloud gives the compliance team the records
they need without having to build custom logging infrastructure themselves.

## Conclusion {#conclusion}

> By building on Pulumi's Automation API, Pulumi Deployments, and Pulumi ESC, Boost turned a serious

Reviewer: ```suggestion
By building on Pulumi's infrastructure platform, Boost turned a serious
```

compliance liability into a self-service capability its developers actually enjoy using. Production
data stays inside Boost's controlled GCP infrastructure, isolated environments spin up in about 15
minutes and tear themselves down on schedule, and the audit trail regulators require is captured

> automatically — all without a ticket queue for the platform team.

Reviewer: ```suggestion
automatically — all without needing to file a request to the platform team.
```
